compiler/semtypes.nim

```diff
@@ -1042,6 +1050,23 @@ proc semBlockType(c: PContext, n: PNode, prev: PType): PType =
 proc semGenericParamInInvocation(c: PContext, n: PNode): PType =
   result = semTypeNode(c, n, nil)
 
+proc semObjectType(c: PContext, n: PNode, t: PType) =
```

Comment:
name this ``semObjectTypeForInheritedGenericInst`` since it's only used for this case.